### DISCUSSION TITLE
Better to have Yes no field as integer or string 1 character?

### DISCUSSION THREAD

**Discussion by TinyTimothy2** [TabularEditor/TabularEditor3#1274] [Data Modeling]
Is it better to have Yes no field as integer or string 1 character Y, N
Is there really a performance and space diff for 1 B row table?
I am thinking of switching back Y/N because it is better and more intuitive in visuals and filters.

**Comment by otykier** [comment]
All that matters is the cardinality of the column. So in theory, there should be no difference between using 0/1 (integer), true/false (boolean) or "Y"/"N" (one character string) or even "Yes"/"No" (multiple character string). Use the option that makes most sense for your model consumers.

**Reply by TinyTimothy2** [reply]
Does the size matter?  Won't characters take up more space which would also take up more memory?

**Reply by otykier** [reply]
Doesn’t matter. String columns are dictionary encoded.

If you’re curious about this stuff, please spend some time educating yourself on the [VertiPaq Engine](https://www.microsoftpressstore.com/articles/article.aspx?p=2449192&seqNum=3), which is the storage engine behind Power BI semantic models and Analysis Services tabular models.
